components/search_results/search_results.tsx

```diff
@@ -0,0 +1,296 @@
+    // Shallow compare for all props except 'results'
+    for (const key in nextProps) {
+        if (!Object.prototype.hasOwnProperty.call(nextProps, key) || key === 'results') {
+            continue;
+        }
+
+        if (nextProps[key] !== props[key]) {
+            return true;
+        }
+    }
+
+    // Here we do a slightly deeper compare on 'results' because it is frequently a new
+    // array but without any actual changes
+    const {results} = props;
+    const {results: nextResults} = nextProps;
+
+    if (results.length !== nextResults.length) {
+        return true;
+    }
+
+    for (let i = 0; i < results.length; i++) {
+        // Only need a shallow compare on each post
+        if (results[i] !== nextResults[i]) {
+            return true;
+        }
+    }
+
+    return false;
+};
+
+interface NoResultsProps {
+    variant: NoResultsVariant,
+    titleValues?: Record<string, JSX.Element|string>
+    subtitleValues?: Record<string, JSX.Element|string>
+}
+
+const defaultProps: Partial<Props> = {
+    isCard: false,
+    isOpened: false,
+    channelDisplayName: '',
+    children: null,
+};
+
+const SearchResults: React.FC<Props> = (props: Props): JSX.Element => {
+    const scrollbars = useRef<Scrollbars>(null);
+
+    useEffect(() => {
+        scrollbars.current?.scrollToTop();
+    }, [props.searchTerms]);
+
+    const handleScroll = (): void => {
+        if (!props.isFlaggedPosts && !props.isPinnedPosts && !props.isSearchingTerm && !props.isSearchGettingMore) {
+            const scrollHeight = scrollbars.current?.getScrollHeight() || 0;
+            const scrollTop = scrollbars.current?.getScrollTop() || 0;
+            const clientHeight = scrollbars.current?.getClientHeight() || 0;
+            if ((scrollTop + clientHeight + GET_MORE_BUFFER) >= scrollHeight) {
+                loadMorePosts();
+            }
+        }
+    };
+
+    const loadMorePosts = debounce(
+        () => {
+            props.getMorePostsForSearch();
+        },
+        100,
+        false,
+        (): void => {},
+    );
+
+    const {
+        results,
+        searchTerms,
+        isCard,
+        isSearchAtEnd,
+        isSearchingTerm,
+        isFlaggedPosts,
+        isSearchingFlaggedPost,
+        isPinnedPosts,
+        isSearchingPinnedPost,
+        isSideBarExpanded,
+        isMentionSearch,
+        isOpened,
+        updateSearchTerms,
+    } = props;
+
+    const noResults = (!results || !Array.isArray(results) || results.length === 0);
+    const isLoading = isSearchingTerm || isSearchingFlaggedPost || isSearchingPinnedPost || !isOpened;
+    const showLoadMore = !isSearchAtEnd && !isFlaggedPosts && !isPinnedPosts;
+
+    let contentItems;
+    let loadingMorePostsComponent;
+
+    let sortedResults = results;
+
+    const titleDescriptor: Record<string, string> = {};
+    const noResultsProps: NoResultsProps = {
+        variant: NoResultsVariant.ChannelSearch,
+    };
+
+    switch (true) {
```

Comment:
Can you replace this block with an if-else instead? I can understand what it does after looking at it for a minute, but I prefer to be more obvious with what code does since other people will have to read and understand it as well.

Same with where you're doing that below

Reply:
I agree with most of your points but I personally find the switch more readable, instead of chaining 5-6 if else statements like it was. I know there might be some performance issues but not in the case of this component.

Reply:
I do agree with @nevyangelova here

Reply:
As someone who now knows what this does, I don't see much difference between
```javascript
if (a) {
} else if (b) {
} else {
}
```
and
```javascript
switch(true) {
case a:
    break;
case b:
    break;
default:
}
```
readability-wise other than the switch being slightly longer because of the breaks (which does seem unnecessary), but I'm more concerned with readability for people who aren't familiar with the code. Since this is an open source project, we need to be careful to make sure that others can understand and work with the code, and that means we have to avoid "clever" stuff that isn't immediately obvious to the reader when there's an easier way to write it.

Reply:
mmh, i know what you mean, but in my opinion this is a very basic pattern in javascript and increases readability by a fair amount. This is cited from an SO-post about performance comparison of `if/else` and `switch`: 

> That's micro optimization and premature optimization, which are evil. Rather worry about readabililty and maintainability of the code in question. If there are more than two `if/else` blocks glued together or its size is unpredictable, then you may highly consider a `switch` statement.

So performance wise there is no difference (`switch` is slightly faster, but almost unmeasurable). And readability is improved with a `switch` statement.

Any opinions on that @nevyangelova?

Reply:
I still hold my opinion about keeping the `switch` statement. The switch statement is faster in most cases when compared to if-else, but significantly faster only when the number of conditions is large. The primary difference in performance between the two is that the incremental cost of an additional condition is larger for if-else than it is for switch. Therefore, our natural inclination to use if-else for a small number of conditions and a switch statement for a larger number of conditions is exactly the right advice when considering performance.

Reply:
Sorry, I meant to continue this conversation sooner since I'm 5/5 that this is the wrong way to do this. I know it's a silly thing to be arguing about, but my main problem with this is that it's making a simple piece of code less obvious for people working on this code, especially those who may be seeing it for the first time. We'll have a lot of new people in that situation as well since this is open source. Switch statements are intended to be used when you're checking if something is one of a number of different values, and the thing where I get caught on is that we're using it to check many different conditions here. It is technically correct this way, but I don't want people to have to look at StackOverflow to understand a conditional statement.

Reply:
I forgot to reply with a review so that this thread doesn't get lost, but I still would like that changed. I'd be up to chat more about it though if you'd like.

Reply:
I can totally understand your reasoning behind it, but I also think that working on open source software should be seen as opportunities to learn new things. I am aligning with the coding guidelines as well. Simple checks (1-2 easy conditions) are best handled with `if-else`, more discrete values should require the usage of a `switch` statement.

It is a different story when you do have very complex conditions, but with these simple ones (all except one are single boolean values) a `switch` should be the obvious choice

Reply:
@hmhealey i did change it to a if/else block. I can relate to your reasoning and don't think arguing about it will bring us anywhere, since it is mostly a matter of personal coding style.